configure.ac

```diff
@@ -2041,18 +2041,18 @@ else
     PYVER=0
   fi
   if test "x$PY3BIN" = xyes; then
-    if test x"$PYOSNAME" = x"nt" -a x"$PYSEPARATOR" = x"\\" -a $PYVER -ge 3; then
```

Comment:
This python configure is rather delicate and this is a massive change in this area. I'd rather not touch it any more unless we get rid of Python 2.7. Is there something not working on Github Actions that requires these changes? I would rather see it in a separate patch with motivating examples and minimal changes if possible.

Reply:
I was not refereeing to python 2.
I did not add any python 2 tests in `win_ci.yml`.
Decisions regarding Python 2 is beyond this PR.

Yes, MSYS2 MINGw64 python have `python-config`.
Somehow the configure.ac decide that all windows do not use `python-config`.
My change is in case we do have `python-config`, use it and skip "windows" probing, and if we do not have `python-config`, use windows special probe.

The change is only on windows and when the python have `python-config`, this is the reason all python tests passed.

I can remove MSYS2 MINGw64 with python 3 test for testing.
If you wish, but it is the only MINGw64 test available at the moment.

Reply:
Revert configure.ac

I add ruby with MingW64 instead.